openedx/core/djangoapps/appsembler/sites/tasks.py

```diff
@@ -61,18 +109,14 @@ def clone_course(source_course_key_string, destination_course_key_string, user_i
         logging.exception(u'Course Clone Error')
         return "duplicate course"
 
-    except SearchIndexingError as search_err:
-        logging.exception(u'Course Clone index Error')
-        return "index error"
-
     # catch all exceptions so we can update the state and properly cleanup the course.
     except Exception as exc:  # pylint: disable=broad-except
         # update state: Failed
         logging.exception(u'Course Clone Error')
 
         try:
             # cleanup any remnants of the course
-            modulestore().delete_course(destination_course_key, user_id)
+            modulestore().delete_course(course_target_id, organization.users.first())
```

Comment:
The task should get the final `user_id` in my opinion instead of trying to calculate it.

```suggestion
            modulestore().delete_course(course_target_id, user_id)
```

Reply:
@OmarIthawi It makes more sense in this way for, the user needs to belong to the org, encapsulating the logic in the function protects someone adding a user from another org to the course, we have had that problem in the past, of users from one org, belonging to courses in other org.